src/test/java/htsjdk/samtools/CRAMIteratorTest.java

```diff
@@ -0,0 +1,50 @@
+package htsjdk.samtools;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.ref.ReferenceSource;
+import htsjdk.samtools.seekablestream.SeekableStream;
+import org.junit.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+
+/**
+ * @author Anton_Mazur@epam.com, EPAM Systems, Inc.
+ *
+ * This test serve for verifying CRAMIterator records validation using strict validation strategy
+ **/
+
+public class CRAMIteratorTest extends HtsjdkTest {
+    private static final File refFile = new File("src/test/resources/htsjdk/samtools/cram/ce.fa");
+    private static final File cramFile = new File("src/test/resources/htsjdk/samtools/cram/ce#supp.3.0.cram");
+    ReferenceSource source = new ReferenceSource(refFile);
+
+    @Test
+    public void shouldNotThrowsExceptionIfCRAMfileContainsInvalidRecords() {
+        SAMRecordIterator cramIter =
+                getCramFileIterator(cramFile, source, ValidationStringency.STRICT);
+
+        Assert.assertTrue(cramIter.hasNext());
+    }
+
+    @Test(expectedExceptions = SAMException.class)
+    public void shouldThrowExceptionIfCRAMFileContainsInvalidRecods() {
+        SAMRecordIterator cramIter =
+                getCramFileIterator(cramFile, source, ValidationStringency.STRICT);
+
+        while (cramIter.hasNext())
+        {
```

Comment:
We don't publish a style guide, but we generally try to put opening braces on the line with the previous statement.

Reply:
One other comment I forgot to make; since this test is expected to throw an exception, it would be a good idea to use a try/finally block to close the reader.